src/components/InAppBrowser/InAppBrowser.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useRef, useState, FC } from 'react';
+import { Linking, TextInput } from 'react-native';
+import WebView from 'react-native-webview';
+import { WebViewNavigation, WebViewNavigationEvent } from 'react-native-webview/lib/WebViewTypes';
+import styled from 'styled-components/native';
+
+// Components
+import Modal from 'components/Modal';
+
+// Utils
+import { useThemeColors, useIsDarkTheme } from 'utils/themes';
+
+// Local
+import AddressBar from './AddressBar';
+import BrowserFloatingActions from './BrowserFloatingActions';
+import BrowserOptionsModal from './BrowserOptionsModal';
+
+const DEFAULT_URL = 'https://www.pillar.fi/';
+
+interface IInAppBrowser {
+  initialUrl?: string | null;
+  iconUrl?: string | null;
+}
+
+const InAppBrowser: FC<IInAppBrowser> = ({ initialUrl = '', iconUrl }) => {
+  const colors = useThemeColors();
+  const isDarkTheme = useIsDarkTheme();
+
+  const webviewRef = useRef<WebView>(null);
+  const urlInputRef = useRef<TextInput>(null);
+
+  const [url, setUrl] = useState(initialUrl);
+  const [urlValue, setUrlValue] = useState(null);
+
+  const [isTyping, setIsTyping] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [canGoBack, setCanGoBack] = useState(false);
+  const [canGoForward, setCanGoForward] = useState(false);
+
+  useEffect(() => {
+    if (!initialUrl || initialUrl === '') setUrl(DEFAULT_URL);
```

Comment:
Could you just check that empty strings evaluate falsely? Then you could just do:

```suggestion
    if (!initialUrl) setUrl(DEFAULT_URL);
```

Reply:
It does indeed evaluate falsely

Reply:
@Yash-Chaubal _that darned javascript!_  😂